rbac/management/role/view.py

```diff
@@ -336,27 +362,13 @@ def destroy(self, request, *args, **kwargs):
             HTTP/1.1 204 NO CONTENT
         """
         validate_uuid(kwargs.get("uuid"), "role uuid validation")
-        role = self.get_object()
-        if role.system or role.platform_default:
-            key = "role"
-            message = "System roles cannot be deleted."
-            error = {key: [_(message)]}
-            raise serializers.ValidationError(error)
+
         try:
             with transaction.atomic():
-                self.delete_policies_if_no_role_attached(role)
```

Comment:
This needed to come after dual write handler reads current state. See code comments below.